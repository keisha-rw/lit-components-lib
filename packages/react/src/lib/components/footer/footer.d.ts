import * as React from 'react';
export declare const PdsFooter: React.FunctionComponent<Partial<{
    variant: "default" | "subtle";
    layoutContainerVariant: "default" | "narrow";
    layoutContainerRemovePadding: "md" | "all" | undefined;
    behavior: "default" | "login" | "super";
    hideContactPhone: boolean;
    hideHelpLink: boolean;
    hideContactLink: boolean;
    loginSupportLink: "https://www.google.com/help";
    termsOfUseLink: string;
    disclosuresLink: String;
    privacyLink: string;
    securityLink: string;
    reportFraudLink: String;
    logoHref: string;
    logoAriaLabel: String;
    hideLogo: boolean;
    hideSocialIcons: boolean;
    hideLegalLinks: boolean;
    legalNavAriaLabel: String;
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Partial<{}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
