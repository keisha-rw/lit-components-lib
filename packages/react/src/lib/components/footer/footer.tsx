import * as React from 'react';
import { PdsFooter as PdsFooterElement } from '@keisha/design-system';
import { PdsFooterProps } from '../../built-component-props';
import { createPdsReactComponent } from '../../create-pds-react-component';

export const PdsFooter = createPdsReactComponent<PdsFooterProps>()({
  tagName: 'pds-footer',
  elementClass: PdsFooterElement,
  react: React,
});
