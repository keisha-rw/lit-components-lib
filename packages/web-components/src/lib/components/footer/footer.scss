@import '../footer-contact-link/footer-contact-link-mixin';

/*------------------------------------*\
 #FOOTER
\*------------------------------------*/

:host {
  display: block;
}

.pds-c-footer {
  // overrides the hover value set for the .pds-c-link--inverted class
  --pds-semantic-text-interactive-inverted-xstrong: var(
    --pds-semantic-text-inverted-default
  );
  // overrides .pds-c-link--inverted color if it is redefined in another theme - NOTE: this approach is only temporary until we implement component level tokens for footer.
  --pds-semantic-text-interactive-inverted-default: var(
    --pds-semantic-text-inverted-default
  );
  --pds-collapsible-border-icon-hover: var(
    --pds-semantic-border-inverted-default
  );
  --pds-link-text-alt: var(--pds-semantic-text-inverted-default);
  --pds-link-text-alt-hover: var(--pds-semantic-text-inverted-default);

  color: var(--pds-semantic-text-inverted-default);
}

.pds-c-footer--subtle {
  --pds-link-text-alt: var(--pds-semantic-text-default);
  --pds-link-text-alt-hover: var(--pds-semantic-text-default);
  --pds-collapsible-border-icon-hover: var(--pds-semantic-border-icon-default);
  --pds-collapsible-link-text: var(--pds-link-text-alt);
  --pds-collapsible-link-text-hover: var(--pds-link-text-alt-hover);

  color: var(--pds-semantic-text-default);
}

/**
 * Styles the fallback content for the slot
 */
slot[name='logo'] {
  * {
    margin-bottom: size(5);
    max-height: size(7.25);
  }
}

.pds-c-footer__logo-link {
  display: block;
  max-width: size(18);

  svg {
    display: block;
    width: 100%;
  }
}

.pds-c-footer--super .pds-c-footer__navigation-area {
  @media only screen and (min-width: $pds-bp-md) {
    margin-top: size(8);
  }
}

.pds-c-footer__extra-links::part(list) {
  white-space: nowrap;
  display: grid;
  grid-template-columns: 1fr;

  @media only screen and (min-width: $pds-bp-md) {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
  }

  @media only screen and (min-width: $pds-bp-lg) {
    display: flex;
    flex-direction: row;
  }
}

.pds-c-footer__login-support {
  padding-bottom: size(3);

  @include pds-semantic-body-default;
}

::slotted([slot='additional-info']) {
  @include pds-semantic-body-default-compact;

  border-top: size(0.125) solid var(--pds-semantic-border-inverted-default);
  padding-top: size(3);
  margin-top: size(6);

  .pds-c-footer--subtle & {
    border-top: size(0.125) solid var(--pds-semantic-border-default);
  }
}

.pds-c-footer__contact-row {
  pds-link::part(link) {
    @include pds-semantic-body-default-compact;
  }
}

slot[name='custom-contact'] {
  .pds-c-footer__contact-row {
    @include footer-contact-link;

    border-bottom: 1px solid var(--pds-semantic-border-inverted-default);

    .pds-c-footer--subtle & {
      border-bottom: 1px solid var(--pds-semantic-border-default);
    }
  }
}

::slotted(pds-list[slot='custom-contact']) {
  @include footer-contact-link;

  border-bottom: 1px solid var(--pds-semantic-border-inverted-default);

  .pds-c-footer--subtle & {
    border-bottom: 1px solid var(--pds-semantic-border-default);
  }
}

.pds-c-footer__copyright-and-social {
  @media only screen and (min-width: $pds-bp-md) {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
  }
}

.pds-c-footer__legal {
  @include pds-semantic-body-sm;

  padding-top: size(3);
}

.pds-c-footer__legal-text-container,
.pds-c-footer__legal-text-container::slotted([slot='legal-text']) {
  display: grid;
  gap: size(1);
}

.pds-c-footer__legal-text {
  margin: 0;
}

.pds-c-footer__legal-nav {
  padding-top: size(3);
  border-bottom: 1px solid var(--pds-semantic-border-inverted-default);
  padding-bottom: size(3);

  pds-link::part(link) {
    @include pds-semantic-body-default-compact;
  }

  .pds-c-footer--subtle & {
    border-bottom: 1px solid var(--pds-semantic-border-default);
  }
}

.pds-c-footer__social-icons,
::slotted(.pds-c-footer__social-icons) {
  place-self: center end;
  padding-right: size(5);
  margin-top: size(4);

  &::part(list) {
    gap: size(3);
    align-items: center;
  }

  // since social icons are now slotted, styling by ::part doesn't work
  --pds-semantic-text-interactive-inverted-xstrong: var(
    --pds-semantic-border-strong
  );
  --pds-link-text-alt-hover: var(--pds-semantic-border-strong);

  @media only screen and (min-width: $pds-bp-md) {
    margin-top: size(3);
    float: right;
  }
}

.pds-c-footer__logo {
  margin-bottom: size(5);

  ::slotted(img),
  ::slotted(svg) {
    max-height: size(7.25);
  }
}
