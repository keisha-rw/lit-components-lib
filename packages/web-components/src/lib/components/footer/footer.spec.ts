import * as path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import { assert, fixture } from '@open-wc/testing';
import { waitFor } from '@testing-library/dom';
import '@testing-library/jest-dom';
import { PdsFooter } from './footer';
import createSnapshots from '../../../../test/utils/snapshots';

initStoryshots({
  storyKindRegex: /Footer/,
  framework: 'web-components',
  configPath: path.join(__dirname, '../../../../.storybook'),
  asyncJest: true,
  test: createSnapshots,
});

describe('Footer unit tests', () => {
  it('is an instance of PdsFooter', async () => {
    const element = await fixture('<pds-footer/>');
    expect(element).toBeInstanceOf(PdsFooter);
  });

  it('generates login support when behavior is set to login', async () => {
    const element = await fixture('<pds-footer behavior="login" />');
    expect(element).toMatchSnapshot();
  });

  it('does not generate login support when behavior is not set to login', async () => {
    const element = await fixture('<pds-footer behavior="notlogin" />');
    expect(element).toMatchSnapshot();
  });

  it('applies subtle markup when variant is subtle', async () => {
    const element = await fixture('<pds-footer variant="subtle" />');
    expect(element).toMatchSnapshot();
  });

  it('generates spanish support when lang is switched to es', async () => {
    const element = await fixture('<pds-footer lang="es" />');
    expect(element).toMatchSnapshot();
  });

  it('is accessible', async () => {
    const element = await fixture('<pds-footer/>');
    await assert.isAccessible(element);
  });

  it('generates the default Company Co logo when behavior is set to super', async () => {
    const element = await fixture(
      `<pds-footer
        variant="default"
        behavior="super"
      ><pds-footer-nav variant="default" behavior="3col"
      ><pds-footer-nav-item label="Our company">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >About us</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Investor relations</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >News room</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Sustainability</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Insights</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item>
    <pds-footer-nav-item label="We're hiring">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Careers</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Global jobs</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Financial professional opportunities</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Internships</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Recent graduates</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item>
    <pds-footer-nav-item label="Other sites">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >For dental providers</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >For financial professionals</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Company Co Asset Management</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item></pds-footer-nav></pds-footer>`,
    );

    expect(
      element.shadowRoot?.querySelector('a.pds-c-footer__logo-link'),
    ).toHaveAttribute('aria-label');

    expect(element).toMatchSnapshot();
  });

  it('adds aria-label attributes to social icons', async () => {
    const element = await fixture(
      `<pds-footer
        variant="default"
      ></pds-footer>`,
    );

    // Select all pds-link elements in the social-icons list.
    const pdsLinks = element.shadowRoot?.querySelectorAll(
      'pds-list[class="pds-c-footer__social-icons"] pds-link',
    );

    // Loop through each pds-link and expect an aria-label on the link tag.
    pdsLinks?.forEach((pdsLink) =>
      expect(pdsLink?.shadowRoot?.querySelector('a')).toHaveAttribute(
        'aria-label',
      ),
    );

    expect(element).toMatchSnapshot();
  });

  it('hides the contact phone, help link and the contact link when the string of none is entered', async () => {
    const element = (await fixture(
      `<pds-footer
        variant="subtle"
        hideContactPhone="true"
        hideHelpLink="true"
        hideContactLink="true"
        behavior="super"
    ><pds-footer-nav variant="default" behavior="3col"
      ><pds-footer-nav-item label="Our company">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >About us</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Investor relations</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >News room</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Sustainability</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Insights</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item>
    <pds-footer-nav-item label="We're hiring">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Careers</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Global jobs</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Financial professional opportunities</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Internships</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Recent graduates</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item>
    <pds-footer-nav-item label="Other sites">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >For dental providers</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >For financial professionals</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Company Co Asset Management</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item></pds-footer-nav></pds-footer>`,
    )) as PdsFooter;
    // Look for the contact us phone link with phone icon and make sure it is not there.
    expect(
      element.shadowRoot?.querySelector('pds-link pds-icon-phone'),
    ).toBeNull();

    // Look for the help topics link with help circle icon and make sure it is not there.
    expect(
      element.shadowRoot?.querySelector('pds-link pds-icon-help-circle'),
    ).toBeNull();

    // Look for the contact us mail icon and make sure it is not there.
    expect(
      element.shadowRoot?.querySelector('pds-link pds-icon-mail'),
    ).toBeNull();
  });

  it('shows the contact phone, help link and the contact link when at least one has a value', async () => {
    const element = (await fixture(
      `<pds-footer
        variant="subtle"
        hideContactPhone="true"
        hideContactLink="true"
        behavior="super"
    ><pds-footer-nav variant="default" behavior="3col"
      ><pds-footer-nav-item label="Our company">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >About us</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Investor relations</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >News room</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Sustainability</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Insights</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item>
    <pds-footer-nav-item label="We're hiring">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Careers</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Global jobs</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Financial professional opportunities</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Internships</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Recent graduates</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item>
    <pds-footer-nav-item label="Other sites">
      <pds-list>
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >For dental providers</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >For financial professionals</pds-link
          ></pds-list-item
        >
        <pds-list-item
          ><pds-link
            variant="inverted"
            href="#"
            >Company Co Asset Management</pds-link
          ></pds-list-item
        >
      </pds-list>
    </pds-footer-nav-item></pds-footer-nav></pds-footer>`,
    )) as PdsFooter;
    expect(element).toMatchSnapshot();
  });

  it('should render default variant collapsible contact links correctly', async () => {
    const element = (await fixture(
      '<pds-footer variant="default" behavior="super" />',
    )) as PdsFooter;

    element.responsiveViewportSize = 'xs';
    await element.updateComplete;

    expect(element.shadowRoot?.querySelector('pds-collapsible')).toBeTruthy();
    expect(
      element.shadowRoot?.querySelector('pds-collapsible'),
    ).toHaveAttribute('variant', 'inverted');
  });

  it('should render subtle variant collapsible contact links correctly', async () => {
    const element = (await fixture(
      '<pds-footer variant="subtle" behavior="super" />',
    )) as PdsFooter;

    element.responsiveViewportSize = 'xs';
    await element.updateComplete;

    expect(element.shadowRoot?.querySelector('pds-collapsible')).toBeTruthy();
    expect(
      element.shadowRoot?.querySelector('pds-collapsible'),
    ).toHaveAttribute('variant', 'strong');
  });

  it('renders a list of custom links with correct label and href', async () => {
    const element = (await fixture(
      `<pds-footer
      variant="subtle"
      behavior="super"
    >
    <pds-list slot="custom-legal-links">
        <pds-list-item>
          <pds-link
            variant="inverted"
            href="#"
            >Link 1</pds-link
          >
        </pds-list-item>
        <pds-list-item>
          <pds-link
            variant="inverted"
            href="#"
            >Link 2</pds-link
          >
        </pds-list-item>
      </pds-list>
    </pds-footer>`,
    )) as PdsFooter;
    await waitFor(
      () => {
        const legalLinks = element.shadowRoot
          ?.querySelector('.pds-c-footer__extra-links')
          ?.querySelectorAll('pds-link');
        expect(legalLinks?.length).toBe(7);
      },
      {
        timeout: 1000,
      },
    );
  });

  it('renders a list of custom legal links', async () => {
    const element = (await fixture(
      `<pds-footer
      variant="subtle"
      behavior="super"
    >
    <pds-list slot="custom-legal-links">
        <pds-list-item>
          <pds-link
            variant="inverted"
            href="#"
            >Link 1</pds-link
          >
        </pds-list-item>
        <pds-list-item>
          <pds-link
            variant="inverted"
            href="#"
            >Link 2</pds-link
          >
        </pds-list-item>
      </pds-list>
    </pds-footer>`,
    )) as PdsFooter;
    await waitFor(
      () => {
        const legalLinks = element.shadowRoot
          ?.querySelector('.pds-c-footer__extra-links')
          ?.querySelectorAll('pds-link');
        expect(legalLinks?.length).toBe(7);
      },
      {
        timeout: 1000,
      },
    );
  });

  it('hides the social icons when hideSocialIcons is set to true', async () => {
    const element = (await fixture(
      `<pds-footer
      variant="subtle"
      behavior="super"
      hideSocialIcons="true"
    />`,
    )) as PdsFooter;
    expect(
      element.shadowRoot?.querySelector('.pds-c-footer__social-icons'),
    ).not.toBeInTheDocument();
  });
});
