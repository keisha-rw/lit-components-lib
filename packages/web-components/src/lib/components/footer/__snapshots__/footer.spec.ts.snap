// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer unit tests adds aria-label attributes to social icons 1`] = `
<pds-footer
  variant="default"
/>
`;

exports[`Footer unit tests applies subtle markup when variant is subtle 1`] = `
<pds-footer
  variant="subtle"
/>
`;

exports[`Footer unit tests does not generate login support when behavior is not set to login 1`] = `
<pds-footer
  behavior="notlogin"
/>
`;

exports[`Footer unit tests generates login support when behavior is set to login 1`] = `
<pds-footer
  behavior="login"
/>
`;

exports[`Footer unit tests generates spanish support when lang is switched to es 1`] = `
<pds-footer
  lang="es"
/>
`;

exports[`Footer unit tests generates the default Company Co logo when behavior is set to super 1`] = `
<pds-footer
  behavior="super"
  variant="default"
>
  <pds-footer-nav
    behavior="3col"
    variant="inverted"
  >
    <pds-footer-nav-item
      label="Our company"
      variant="inverted"
    >


      <pds-list
        spacing="sm"
      >


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            About us
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Investor relations
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            News room
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Sustainability
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Insights
          </pds-link>
        </pds-list-item>


      </pds-list>


    </pds-footer-nav-item>


    <pds-footer-nav-item
      label="We're hiring"
      variant="inverted"
    >


      <pds-list
        spacing="sm"
      >


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Careers
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Global jobs
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Financial professional opportunities
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Internships
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Recent graduates
          </pds-link>
        </pds-list-item>


      </pds-list>


    </pds-footer-nav-item>


    <pds-footer-nav-item
      label="Other sites"
      variant="inverted"
    >


      <pds-list
        spacing="sm"
      >


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            For dental providers
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            For financial professionals
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Company Co Asset Management
          </pds-link>
        </pds-list-item>


      </pds-list>


    </pds-footer-nav-item>
  </pds-footer-nav>
</pds-footer>
`;

exports[`Footer unit tests shows the contact phone, help link and the contact link when at least one has a value 1`] = `
<pds-footer
  behavior="super"
  hidecontactlink="true"
  hidecontactphone="true"
  variant="subtle"
>
  <pds-footer-nav
    behavior="3col"
    variant="strong"
  >
    <pds-footer-nav-item
      label="Our company"
      variant="strong"
    >


      <pds-list
        spacing="sm"
      >


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            About us
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Investor relations
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            News room
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Sustainability
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Insights
          </pds-link>
        </pds-list-item>


      </pds-list>


    </pds-footer-nav-item>


    <pds-footer-nav-item
      label="We're hiring"
      variant="strong"
    >


      <pds-list
        spacing="sm"
      >


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Careers
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Global jobs
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Financial professional opportunities
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Internships
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Recent graduates
          </pds-link>
        </pds-list-item>


      </pds-list>


    </pds-footer-nav-item>


    <pds-footer-nav-item
      label="Other sites"
      variant="strong"
    >


      <pds-list
        spacing="sm"
      >


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            For dental providers
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            For financial professionals
          </pds-link>
        </pds-list-item>


        <pds-list-item
          size="md"
        >
          <pds-link
            href="#"
            variant="inverted"
          >
            Company Co Asset Management
          </pds-link>
        </pds-list-item>


      </pds-list>


    </pds-footer-nav-item>
  </pds-footer-nav>
</pds-footer>
`;
